mitosheet/mitosheet/sheet_functions/types/utils.py

```diff
@@ -108,28 +108,52 @@ def put_nan_indexes_back(series: pd.Series, original_index: pd.Index) -> pd.Seri
 
 def get_datetime_format(string_series: pd.Series) -> Optional[str]:
     """
-    Given a series of datetime strings, detects if the format is MM-DD-YYYY,
-    which is the most common format that pandas does not default to.
-
-    In the future, we can extend this to detect other formats. Returns None
-    if infer_datetime_format is good enough!
+    Given a series of datetime strings, guesses the most likely date format.
     """
     try:
-        # If we can convert all non null inputs, then we assume that pandas
-        # is guessing the input correctly
+        # Import log function here to avoid circular import
+        from mitosheet.telemetry.telemetry_utils import log
+
+        # If we can convert all non null inputs, then we assume we guessed correctly
         non_null_inputs = string_series[~string_series.isna()]
+
+        # First try letting pandas guess the correct datetime
         converted = pd.to_datetime(non_null_inputs, errors='coerce', infer_datetime_format=True)
-        if converted.isna().sum() > 0:
-            raise Exception("Non full conversion")
+        if converted.isna().sum() == 0:
+            return None
+
+        # Then we try the most popular datetime formats
+        sample_string_datetime = string_series[string_series.first_valid_index()]
+        FORMATS = [
+            '%m{s}%d{s}%Y', 
+            '%d{s}%m{s}%Y', 
+            '%Y{s}%d{s}%m', 
+            '%Y{s}%m{s}%d', 
+            '%m{s}%d{s}%Y %H:%M:%S', 
+            '%d{s}%m{s}%Y  %H:%M:%S', 
+            '%Y{s}%d{s}%m  %H:%M:%S', 
+            '%Y{s}%m{s}%d  %H:%M:%S'
+        ]
+        SEPERATORS = ['/', '-', '.', ':', ' ', '']
```

Comment:
Yeah makes sense. Since we don't have other formats that we want to to add, I'll just leave it as is for now. We can call the variable `MOST_COMMON_FORMATS` and just run through those checks first.